### PR TITLE
Tests for subscribe and better handling of escaped fields

### PR DESCRIPTION
**Pull request by sgaestel**

### CODE REVIEW

File: lib/xml.js [hunk 1]
```diff
@@ -73,22 +73,22 @@ var tag = function (name, prefix, value, attribs) {
 };
 
 var cdata = function(str) {
-  if (common.isString(str) && str.trim().length > 0) {
+  if (!_.isUndefined(str) && !_.isNull(str)) {
```
Comment (arenoux) [inline lib/xml.js]: Why did you update this line ?
Don't we wanna add a CDATA tag only for strings ?

File: test/hot/subscribe.test.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    t.ifError(err && err.message || err);
+    t.ok(z, 'expect valid \'z\' client');
+    if (err) return;
+    z.subscribe(fields, function(err, result) {
+      t.ifError(err && err.message || err);
+      log.debug(z.client._client.lastRequest);
+      log.debug(z.client._client.lastResponse);
+      log.debug(result);
+      t.ok(Array.isArray(result),           'result is an array');
+      t.equal(result.length, 1,             'result has an array length of 2');
```
Comment (arenoux) [inline test/hot/subscribe.test.js]: You want result to be an array of 2 elements, but you you are testing `t.equal(result.length, 1) :p

File: test/hot/subscribe.test.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      t.ok(Array.isArray(result[0].Errors), 'result contains errors array');
+      t.equal(result[0].Errors.length, 2,   'has only 1 result[].Errors[] record');
```
Comment (arenoux) [inline test/hot/subscribe.test.js]: Like in above lines: expected value and message are not consitent

File: lib/xml.js (hunk 1 above)
Comment (sgaestel) [inline lib/xml.js]: I call the cdata function for custom fields, it used to be called only for userFields.
I would suggest maybe calling it systematically and transforming it like this :

```
var cdata = function(value) {
    if (!_.isUndefined(value) && !_.isNull(value)) {
        return '<![CDATA[' + value + ']]>';
    } else {
        return value;
    }
};
```

File: test/hot/subscribe.test.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  t.plan(8);
+  var numRecords = 2;
+
+  var payload = [];
+  for (var i = 0; i < numRecords; i++) {
+    payload.push({
+        Account: {
+            AccountNumber: i,
+            AutoPay: false,
+            Batch: 'this is gonnay fail',
```
Comment (arenoux) [inline test/hot/subscribe.test.js]: I think we need to surround these 3 first keys with simple quotes in order to keep code consistency.
(Same remark for BillToContact and SubscriptionData fields)

File: test/hot/subscribe.test.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            'BillCycleDay': 1,
+            'Currency': 'USD',
+            'Name': 'John "test" Smith ' + i,
+            'PaymentTerm': 'Due Upon Receipt',
+            'SubAccounts__c': '',
+            'UUIDLookUp__c': '',
+            'Status': 'Draft'
+        },
+        BillToContact: {
+            Address1: 'My Street',
```
Comment (arenoux) [inline test/hot/subscribe.test.js]: Do we really need all these fields?
If not, maybe we should keep only necessary fields in order to reduce this object size in tests

File: test/hot/subscribe.test.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+var test  = require('tape');
+var zuora  = require('../../zuora');
+var config = require('../../etc/config.test.json');
+var log    = require('../../lib/logger');
+
+log.level(log.bunyan.DEBUG);
```
Comment (arenoux) [inline test/hot/subscribe.test.js]: I think that we don't need this line

File: lib/xml.js [hunk 7]
```diff
@@ -143,7 +121,7 @@ function convert (method, object, objectType) {
         // Finally, the end value is escaped if one from the userFields.
```
Comment (arenoux) [inline lib/xml.js]: Comment to be updated :p

**Comment by arenoux** [comment]
Good for me :+1:

**Pull request merged**